### ISSUE TITLE
small suggestions

### ISSUE THREAD

**Issue by AccidentallyOnPurpose** [Kornilov1/AI-Dungeon-2-Anons-modded-Versions-Colab#155]
some suggestion for the notebook:

I would make a PR but I'm not sure where the latest version of the notebook is.

If people run the download cell twice they end up in a subdirectory and download the model twice. To avoid that

- `%cd AIDungeon` =>  `%cd $root_dir/AIDungeon`
- `%cd Clover-Edition` =>  `%cd $root_dir/Clover-Edition`

Suggestion 2. If you still have this

```
!pip install regex
!pip install profanityfilter
!pip install playsound
!pip install func_timeout
!pip install tracery
!pip install cryptography
```

perhaps remove it, because it's handled when you install a package requirements.txt anyway. 

Thanks for all your work on this! I'm learning lots about colab and bash by going through this notebook.

**Comment by AccidentallyOnPurpose** [comment]
Suggestion 3. You could print the commit hash a user is using to help you troubleshoot. You can also look at the date of the latest git commit to work out how out of date their code is.

E.g.

```
commit_hash=$(git log --pretty=format:'%h' -n 1)
commit_date=$(git log --pretty=format:'%cd' -n 1)
echo "You are using version https://${repository}/commit/${commit_hash} from ${commit_date}"
```

**Comment by machiavel23** [comment]
thanks for those informations, it'll be handy. for the requirements, we already switched to pip intall -r requirements.txt in the beta, which is the usual dev branch before updating to master.

concerning the commit, i was actually thinking to clone only the master branch with this line
git clone --branch master --depth 1

because for now it's just git clone and it seems to get all the branches, which is unoptimized imo.

and do the same for the git fetch and the git pull, but I'm not quite sure if it is the correct way as I'm still new to git. If you could advise me on that, it'll be a great help.

By the way, I've heard you're the onne who did most of the work on PyTorch, right? Good job if it's that, I'm impressed!

Honestly though, even if I like the idea to see the exact commit actually used, I fear that showing too much precise informations would make the retards cry again :/ The all-in-one cell has become a reality mostly because the majority of users weren't able to follow the few steps we had when the colab had multi cells, and I lack of time to stay tuned with the latest changes of the forks. Stil the idea is quite interesting, I'll let Kornilov see the discussion and we'll talk about it.

**Comment by VBPXKSMI** [comment]
I agree with evertything said above.
Concerning the last paragraph, it also worries me. Brainlets may get confused the more info is displayed in their screens.
Still, the idea of having more accurate troubleshoutting sounds appealling and we should consider it. 
Since I lurk (up till know at least) any and all threads made, I usually get what the problems are, but this might not be possible in the future. 
Either way, we should also add a message along the lines of "If you can't figure out how to solve the problem, post this text in the most recent AI Dungeon thread and ask for advice", this way we can use the collective knowledge of Anons to get the issue solved faster and then we only need to make the necessary changes here.

**Comment by AccidentallyOnPurpose** [comment]
@machiavel23 I've used git for years but I still don't feel like I "know" git if you know what I mean. That said " clone only the master branch" sound like a great idea, I've noticed the slow down of cloning everything.

@machiavel23 yup I did the clover pytorch branch, thanks!

> showing too much precise informations would make the retards cry again :/ 

True, I see what you guys mean, you've convinced me. People see a deprecation notice or something. Then it's all HALP computer is wrong, what i do.

**Comment by machiavel23** [comment]
yep, especially without providing any useful info like colab version or offline, what repo, settings... damn brainlets. hopefully, it's not redditards, but it's still frutrating sometimes

also @Kornilov1 , I've already implemented big error messages to copy paste to the thread, at least to know at which step the process bugs. like

-----------------------------------------------------------
ERROR : MODEL FOLDER MISSING POST INSTALL
-----------------------------------------------------------

It's very basic but i hope big enough for even the most retarded of the retards...

**Comment by machiavel23** [comment]
@AccidentallyOnPurpose @Kornilov1 
actually, in the latest beta, the game gets downloaded and installed directly in gdrive to avoid too much %cd, and mainly, the mv. seems to work. the pip are read from the requirements.txt right after git clone (for initial setup) and right before play.py (for replaying after initial setup).

for git only master, the first idea was to put master by default and give a selector to choose the branch, but i have finally retracted myself to do that because retards . in any cases, I will wait for the feedbacks of the new installation before trying something for git only master, don't want to screw the whole thing up.

**Comment by VBPXKSMI** [comment]
Up till now, all good additions. When the beta is ready I will merge it.

**Comment by machiavel23** [comment]
so it seems there's a big problem with the new installation caused by the virtual environment location causing aria2 (the toole used to launch .torrent files) to malfunction, resulting in a never starting torrenting of the model files. I revert to the usual installation method for now (the one that need !mv) and will implement a small feedback on the output for the mv, because the --verbose parameter of mv tends to create lines too fast and freezes/crashes some browsers on potatoes machines (like mine actually)

**Comment by machiavel23** [comment]
Well... I actually managed to do it...

https://colab.research.google.com/github/Kornilov1/AI-Dungeon-2-Anons-modded-Versions-Colab/blob/beta/Open_CYOAI_Colab_1.22_BETA.ipynb

**Comment by VBPXKSMI** [comment]
> Well... I actually managed to do it...
> 
> https://colab.research.google.com/github/Kornilov1/AI-Dungeon-2-Anons-modded-Versions-Colab/blob/beta/Open_CYOAI_Colab_1.22_BETA.ipynb

@machiavel23 I assume you were the one that shared this Colab in the thread: https://colab.research.google.com/github/biplobsd/Google-Colab-CloudTorrent/blob/master/torrentTOmega_gdrive.ipynb

If you haven't, take a look at how they collapsed the sections upon opening the Colab, we could implement that. 
Also, using .gifs or hell even videos on the Colab explaining how or what to do would save us from putting a wall of text everywhere.

EDIT: upon looking at their GitHub, unless I'm mistaken (which I could be), they seem to have used the extensions you once talked about. We should test if they could work even when copying the Colab to Drive.

**Comment by machiavel23** [comment]
I don't think the extension are needed for that. take a look at the file, pure code style
https://raw.githubusercontent.com/biplobsd/Google-Colab-CloudTorrent/master/torrentTOmega_gdrive.ipynb
and ours
https://raw.githubusercontent.com/Kornilov1/AI-Dungeon-2-Anons-modded-Versions-Colab/beta/Open_CYOAI_Colab_1.22_BETA.ipynb

apparently, we just need to add the sections ID in "collapsed_sections": [] at the beginning. I guess the extensions add a menu for that. but a manual edit should work too. trying now

**Comment by machiavel23** [comment]
so, edited it directly in github, worked perfectly. but if you change the filename, you have also two references to change in the file for it to load in colab. Glad that ctrl+f works xD. check 1.22.3 BETA

**Comment by VBPXKSMI** [comment]
What references? 
The usual ones at the beggining?:
>Open CYOAI Colab v1.22.3 BETA.ipynb
>https://colab.research.google.com/github/Kornilov1/AI-Dungeon-2-Anons-modded-Versions-Colab/blob/beta/Open_CYOAI_Colab_1.22.3_BETA.ipynb

Otherwise I don't see any other reference.

**Comment by machiavel23** [comment]
{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "name": "Open CYOAI Colab v1.22.3 BETA.ipynb",
      "provenance": [],
      "collapsed_sections": [
        "7dRi_BDWErNf",
        "0FBTZwEOzKQ1",
        "9OyOAgpo0XfU",
        "xn0bEX3R8IMC",
        "gmioGz252Ghx",
        "88pDH9FP0TCz",
        "nqgPglYzMeek",
        "9mO5XY3HOBKN",
        "zITeSYmJvBp4",
        "nJMePPmJvBp7",
        "ooL1ozRUgO5r"
      ],
      "machine_shape": "hm",
      "toc_visible": true,
      "include_colab_link": true
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "accelerator": "GPU"
  },
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/Kornilov1/AI-Dungeon-2-Anons-modded-Versions-Colab/blob/beta/Open_CYOAI_Colab_1.22.3_BETA.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"


there's one at the top, and one other a bit later, as you can see above. both must be the same to avoid a colab glitch when directly editing the file as raw

**Comment by VBPXKSMI** [comment]
Man, I must be blind because I simply don't see anything out of the ordinary. Just seeing the usual Colab names being different, alonside the new section with each part that is collapsed.

Can you highlight these references for me?


Incidentally, I edited the collapsable part a bit because I thought the "About" part should be always visible at first,  the rest of the sections I left them collapsed but visible as compared to be "over"-collapse at start.

**Comment by machiavel23** [comment]
at the top
"name": "Open CYOAI Colab v1.22.3 BETA.ipynb",

and at the bottom of my quote
"https://colab.research.google.com/github/Kornilov1/AI-Dungeon-2-Anons-modded-Versions-Colab/blob/beta/Open_CYOAI_Colab_1.22.3_BETA.ipynb"

these both must have the same .ipynb name when you edit the file raw (like in notepad++), if you change the filename. Like, if I want to save the file under the name "test.ipynb" after editing it with notepad++, i'll have to manually change `"name": "Open CYOAI Colab v1.22.3 BETA.ipynb"`, to `"name": "test.ipynb"`, and `"https://colab.research.google.com/github/Kornilov1/AI-Dungeon-2-Anons-modded-Versions-Colab/blob/beta/Open_CYOAI_Colab_1.22.3_BETA.ipynb"` to `"https://colab.research.google.com/github/Kornilov1/AI-Dungeon-2-Anons-modded-Versions-Colab/blob/beta/test.ipynb"`, otherwise it will result in an error when trying to open it in colab

**Comment by VBPXKSMI** [comment]
Ahhh I already new of that, that's why I keep asking you that I didn't see anything out of the ordinary.
Still, thanks for the explanation, sorry for wasting your time.

By the way, when renaming the Colab link like in
>"https://colab.research.google.com/github/Kornilov1/AI-Dungeon-2-Anons-modded-Versions-Colab/blob/beta/Open_CYOAI_Colab_1.22.3_BETA.ipynb"

I suggest that instead of "Colab 1.22.3 BETA" you put just something like "Colab BETA" so you won't have to change the link whith each version that comes out, also it would help me from having to change the Beta Colab link from the OP each time. 
But if you want to have more than 1 version of the Colab at the same time then just create a new file with a different name and Colab link, like I did with the "Old" and "current" main Colabs in the master branch.

tl;dr
The name of the Colab do change it, just not the Colab link name. But that's just my opinion.

**Comment by machiavel23** [comment]
ok, will do that

**Comment by machiavel23** [comment]
if the greasymonkey script helps avoid the disconnect glitch, I suggest adding a link to it on the colab and then close this issue. It's pretty obvious now we can't really do anything against it by ourselves.

**Comment by VBPXKSMI** [comment]
I would like a bit more feedback before doing that honestly, although some reported it works I want to be sure.

**Comment by machiavel23** [comment]
I just checked the content of the script and tested it with my firefox on tampermonkey extension, it doesn't work. I tried to adapt the small function that can be found on google and make it a userscript, but it needs some tweaks too, and apparently, if it works, it doesn't print anything in the console browser. I think by cross-referring the buggy greasymonkey script and the mini function, we should be able to build a simple javascript function that will correctly work on greasemonkey and tampermonkey. It doesn't sound too complicated, but again, no time. putting links for reference:

https://greasyfork.org/en/scripts/386556-google-colab-auto-reconnector
https://medium.com/@shivamrawat_756/how-to-prevent-google-colab-from-disconnecting-717b88a128c0

During my test, it also appeared that the usual javascript functions `getElementById()` and `querySelector()` used to retrieve an html element by its id return null instead of finding the connect button id, same apparently goes for `getElementByClassName()`. It can be because the script run too soon (and so the button is not yet existent in the page) or because the code is not searching the good id.

In any case, it'll require quite some javascript knowledge to build a userscript, but it can definitely be done.
